asr-python-adapter-finetune-am-conformer-nemo.ipynb

```diff
@@ -0,0 +1,186 @@
+   "source": [
+    "## NVIDIA Riva Overview\n",
+    "\n",
+    "NVIDIA Riva is a GPU-accelerated SDK for building speech AI applications that are customized for your use case and deliver real-time performance. <br/>\n",
+    "Riva offers a rich set of speech and natural language understanding services such as:\n",
+    "\n",
+    "- Automated speech recognition (ASR)\n",
+    "- Text-to-Speech synthesis (TTS)\n",
+    "- A collection of natural language processing (NLP) services, such as named entity recognition (NER), punctuation, and intent classification.\n",
+    "\n",
+    "In this tutorial, we will customize a Riva ASR acoustic model (Conformer) with Adapter modules, using the NeMo Toolkit. <br> \n",
+    "To understand the basics of Riva ASR APIs, refer to [Getting started with Riva ASR in Python](https://github.com/nvidia-riva/tutorials/blob/stable/asr-python-basics.ipynb). <br>\n",
+    "\n",
+    "For more information about Riva, refer to the [Riva developer documentation](https://developer.nvidia.com/riva)."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Neural Module (NeMo) Toolkit\n",
+    "NVIDIA Neural Module (NeMo) Toolkit is an open-source framework for building, training, and fine-tuning GPU-accelerated speech AI and natural language understanding (NLU) models with a simple Python interface. Developers, researchers, and software partners building intelligent conversational AI applications and services, can bring their own data to fine-tune pre-trained models instead of going through the hassle of training the models from scratch."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# ASR with Adapters\n",
+    "\n",
+    "The following tutorial will heavily reference the [NeMo tutorial - ASR Domain Adaptation with Adapters](https://colab.research.google.com/github/NVIDIA/NeMo/blob/stable/tutorials/asr/asr_adapters/ASR_with_Adapters.ipynb)\n",
+    "\n",
+    "We advise to keep both tutorials open side by side to refer the contents effectively by using the **Table of Contents**."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# What are Adapters?\n",
+    "\n",
+    "Adapters are trainable neural network modules that are attached to pretrained models, such that we freeze the weights of the original model and only train the adapter parameters. This reduces the amount of data required to customize a model substantially, while imposing the limitation that the model's vocabulary cannot be changed.\n",
+    "\n",
+    "In short,\n",
+    "\n",
+    "- Adapter modules form a residual bridge over the output of each layer they adapt, such that the model's original performance is not lost. \n",
+    "- The original parameters of the model are frozen in their entirety - so that we can recover the original model by disabling all adapters.\n",
```

Comment:
```suggestion
    "- The original parameters of the model are frozen in their entirety, so that we can recover the original model by disabling all adapters.\n",
```